### ISSUE TITLE
ddtrace 2.8.0 crashes with langchain 

### ISSUE THREAD

**Issue by loicgasser** [DataDog/dd-trace-py#8980]
### Summary of problem

Version 2.8.0 of ddtrace crashes when combined with version 0.1.16 of langchain.

### Which version of dd-trace-py are you using?

Problem observed in 2.8.0
Problem doesn't exist in 2.7.4

### Which version of pip are you using?

pip 23.3.2

### Which libraries and their versions are you using?

<details>
  <summary>`pip freeze`</summary>
aioboto3==12.3.0
aiobotocore==2.11.2
aiohttp==3.9.3
aioitertools==0.11.0
aiosignal==1.3.1
annotated-types==0.6.0
anyio==4.3.0
astroid==3.1.0
async-timeout==4.0.3
attrs==23.2.0
boto3==1.34.34
botocore==1.34.34
bytecode==0.15.1
cattrs==23.2.3
certifi==2024.2.2
cfgv==3.4.0
charset-normalizer==3.3.2
click==8.1.7
coverage==7.4.4
dataclasses-json==0.6.4
ddsketch==3.0.1
ddtrace==2.8.0
Deprecated==1.2.14
dill==0.3.8
distlib==0.3.8
distro==1.9.0
envier==0.5.1
fastapi==0.110.1
filelock==3.13.1
frozenlist==1.4.1
greenlet==3.0.3
gunicorn==21.2.0
h11==0.14.0
httpcore==1.0.4
httpx==0.27.0
httpx-sse==0.4.0
identify==2.5.35
idna==3.6
importlib-metadata==6.11.0
iniconfig==2.0.0
isort==5.13.2
jmespath==1.0.1
jsonpatch==1.33
jsonpointer==2.4
langchain==0.1.16
langchain-community==0.0.32
langchain-core==0.1.42
langchain-openai==0.1.3
langchain-text-splitters==0.0.1
langserve==0.0.51
langsmith==0.1.25
libai==0.2.11
libcore==0.1.2
libfiscalyear==0.1.4
marshmallow==3.21.1
mccabe==0.7.0
multidict==6.0.5
mypy-extensions==1.0.0
nodeenv==1.8.0
numpy==1.26.4
openai==1.17.1
opentelemetry-api==1.23.0
orjson==3.9.15
packaging==23.2
pandas==2.2.1
platformdirs==4.2.0
pluggy==1.4.0
pre-commit==3.7.0
protobuf==5.26.0
pydantic==2.6.4
pydantic_core==2.16.3
pylint==3.1.0
pytest==8.1.1
pytest-asyncio==0.23.6
pytest-cov==4.1.0
python-dateutil==2.9.0.post0
python-dotenv==1.0.1
pytz==2024.1
PyYAML==6.0.1
redis==5.0.3
regex==2023.12.25
requests==2.31.0
ruff==0.3.7
s3transfer==0.10.1
-e git+ssh://git@github.com/KeenStrategy/service-ai.git@fb82a2d1f28334d41cb524e6d49401c8dedf9641#egg=service_ai
six==1.16.0
sniffio==1.3.1
SQLAlchemy==2.0.28
sqlparse==0.4.4
sse-starlette==1.8.2
starlette==0.37.2
structlog==24.1.0
tenacity==8.2.3
tiktoken==0.6.0
tomlkit==0.12.4
tqdm==4.66.2
typing-inspect==0.9.0
typing_extensions==4.10.0
tzdata==2024.1
urllib3==2.0.7
uvicorn==0.29.0
virtualenv==20.25.1
wrapt==1.16.0
xmltodict==0.13.0
yarl==1.9.4
zipp==3.18.1
</details>

### How can we reproduce your problem?

add this code in a `main.py` file and use this command `ddtrace-run gunicorn main:app --timeout 0 --workers 4 --worker-class uvicorn.workers.UvicornWorker --bind 0.0.0.0:8000`

```python
from fastapi import FastAPI
from langchain.prompts import PromptTemplate, load_prompt

app = FastAPI()


@app.get("/")
async def root():
    return {"message": "Hello World"}
```

### What is the result that you get?

```
Traceback (most recent call last):
  File "/home/loicgasser/.cache/pypoetry/virtualenvs/service-ai-YK2KbUEN-py3.11/lib/python3.11/site-packages/gunicorn/arbiter.py", line 609, in spawn_worker
    worker.init_process()
  File "/home/loicgasser/.cache/pypoetry/virtualenvs/service-ai-YK2KbUEN-py3.11/lib/python3.11/site-packages/uvicorn/workers.py", line 68, in init_process
    super().init_process()
  File "/home/loicgasser/.cache/pypoetry/virtualenvs/service-ai-YK2KbUEN-py3.11/lib/python3.11/site-packages/gunicorn/workers/base.py", line 134, in init_process
    self.load_wsgi()
  File "/home/loicgasser/.cache/pypoetry/virtualenvs/service-ai-YK2KbUEN-py3.11/lib/python3.11/site-packages/gunicorn/workers/base.py", line 146, in load_wsgi
    self.wsgi = self.app.wsgi()
                ^^^^^^^^^^^^^^^
  File "/home/loicgasser/.cache/pypoetry/virtualenvs/service-ai-YK2KbUEN-py3.11/lib/python3.11/site-packages/gunicorn/app/base.py", line 67, in wsgi
    self.callable = self.load()
                    ^^^^^^^^^^^
  File "/home/loicgasser/.cache/pypoetry/virtualenvs/service-ai-YK2KbUEN-py3.11/lib/python3.11/site-packages/gunicorn/app/wsgiapp.py", line 58, in load
    return self.load_wsgiapp()
           ^^^^^^^^^^^^^^^^^^^
  File "/home/loicgasser/.cache/pypoetry/virtualenvs/service-ai-YK2KbUEN-py3.11/lib/python3.11/site-packages/gunicorn/app/wsgiapp.py", line 48, in load_wsgiapp
    return util.import_app(self.app_uri)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/loicgasser/.cache/pypoetry/virtualenvs/service-ai-YK2KbUEN-py3.11/lib/python3.11/site-packages/gunicorn/util.py", line 371, in import_app
    mod = importlib.import_module(module)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/loicgasser/.pyenv/versions/3.11.3/lib/python3.11/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1206, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1178, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1149, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "/home/loicgasser/.cache/pypoetry/virtualenvs/service-ai-YK2KbUEN-py3.11/lib/python3.11/site-packages/ddtrace/internal/module.py", line 220, in _exec_module
    self.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/home/loicgasser/Workspace/service-ai/main.py", line 2, in <module>
    from langchain.prompts import PromptTemplate, load_prompt
  File "/home/loicgasser/.cache/pypoetry/virtualenvs/service-ai-YK2KbUEN-py3.11/lib/python3.11/site-packages/ddtrace/vendor/wrapt/importer.py", line 177, in _exec_module
    notify_module_loaded(module)
  File "/home/loicgasser/.cache/pypoetry/virtualenvs/service-ai-YK2KbUEN-py3.11/lib/python3.11/site-packages/ddtrace/vendor/wrapt/decorators.py", line 470, in _synchronized
    return wrapped(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/loicgasser/.cache/pypoetry/virtualenvs/service-ai-YK2KbUEN-py3.11/lib/python3.11/site-packages/ddtrace/vendor/wrapt/importer.py", line 136, in notify_module_loaded
    hook(module)
  File "/home/loicgasser/.cache/pypoetry/virtualenvs/service-ai-YK2KbUEN-py3.11/lib/python3.11/site-packages/ddtrace/_monkey.py", line 178, in on_import
    imported_module.patch()
  File "/home/loicgasser/.cache/pypoetry/virtualenvs/service-ai-YK2KbUEN-py3.11/lib/python3.11/site-packages/ddtrace/contrib/langchain/patch.py", line 778, in patch
    wrap("langchain_pinecone", "PineconeVectorStore.similarity_search", traced_similarity_search(langchain))
  File "/home/loicgasser/.cache/pypoetry/virtualenvs/service-ai-YK2KbUEN-py3.11/lib/python3.11/site-packages/ddtrace/vendor/wrapt/wrappers.py", line 882, in wrap_function_wrapper
    return wrap_object(module, name, FunctionWrapper, (wrapper,))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/loicgasser/.cache/pypoetry/virtualenvs/service-ai-YK2KbUEN-py3.11/lib/python3.11/site-packages/ddtrace/vendor/wrapt/wrappers.py", line 822, in wrap_object
    (parent, attribute, original) = resolve_path(module, name)
                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/loicgasser/.cache/pypoetry/virtualenvs/service-ai-YK2KbUEN-py3.11/lib/python3.11/site-packages/ddtrace/vendor/wrapt/wrappers.py", line 777, in resolve_path
    __import__(module)
ModuleNotFoundError: No module named 'langchain_pinecone'
```

### What is the result that you expected?

ddtrace doesn't make my application crash.

**Comment by emmettbutler** [comment]
Thanks for the report @loicgasser. We'll check it out.

cc @Yun-Kim

**Comment by Yun-Kim** [comment]
Hi @loicgasser, thanks for reaching out! We've fixed this bug in https://github.com/DataDog/dd-trace-py/pull/8839 and waiting to have it released on `ddtrace==2.8.2/2.9.0`. In the meantime, one workaround is to have `langchain-openai, langchain-pinecone` installed on your environment.
